src/app/service/game-control.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GameControlService {
+  private currentPlayer: BehaviorSubject<string> = new BehaviorSubject<string>(
+    'circle'
+  );
+  private circle: BehaviorSubject<number[]> = new BehaviorSubject<number[]>([]);
```

Comment:
Do we need both streams? Isn't it enough to keep the game state in one stream? It would also solve players' turns - you could check last player :)